### PR TITLE
LUCENE-9951: Add InfoStream to ReplicationService

### PR DESCRIPTION
**Pull request by ChristophKaser**
An InfoStream is added to the ReplicationService (similar to the ReplicationClient) to allow debugging replication issues

# Description

Adds InfoStream to ReplicationService to facilitate debugging

#10990

# Solution

# Tests

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/lucene/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Lucene maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.

### CODE REVIEW

File: lucene/replicator/src/java/org/apache/lucene/replicator/http/ReplicationService.java [hunk 1]
```diff
@@ -183,6 +203,17 @@ public void perform(HttpServletRequest req, HttpServletResponse resp)
...
     } catch (Exception e) {
+      if (infoStream.isEnabled(INFO_STREAM_COMPONENT)) {
+        final StringWriter sw = new StringWriter();
+        sw.append("an error occurred during replication service call (");
+        sw.append(req.getRequestURI());
+        if (req.getQueryString() != null) {
+          sw.append('?').append(req.getQueryString());
+        }
+        sw.append("): ");
+        e.printStackTrace(new PrintWriter(sw));
```
[the remark this answers is not in the document]
Comment (ChristophKaser) [reply lucene/replicator/src/java/org/apache/lucene/replicator/http/ReplicationService.java:221]: There is no sensitive information in a replication request
@muse-dev ignore

**Comment by mikemccand** [comment]
Thanks @ChristophKaser and sorry for this very late reply!  I like this idea -- Replication is so tricky to debug.  This now has conflicts unfortunately -- do you want to refresh the PR to resolve them?

**Comment by ChristophKaser** [comment]
@mikemccand Thank you for looking at the patch! However it is a bit hard to refresh this PR - after all, the http servlet based replication mechanism has been removed from lucene in PR #12038.
I have yet to look at the 'new' replication mechanism and port my code to it, so I can't really say whether it would benefit from adding an InfoStream to it ATM.

**Comment by mikemccand** [comment]
OK thank you for bringing closure @ChristophKaser.

**Pull request closed without merging**